src/app/games/diam/tests/diam.component.spec.ts

```diff
@@ -0,0 +1,213 @@
+        ], 4);
+        testUtils.setupState(state);
+        // when clicking on A1 and then on the last column
+        // then the move is legal
+        await testUtils.expectClickSuccess('#click_0_1');
+        const move: DiamMove = DiamMoveShift.fromRepresentation(new Coord(0, 2), 'counterclockwise');
+        await testUtils.expectMoveSuccess('#click_7', move);
+    }));
+    it('should forbid shift of more than one space', fakeAsync(async() => {
+        // given a state
+        const state: DiamState = DiamState.fromRepresentation([
+            [__, __, __, __, __, __, __, __],
+            [__, __, __, __, __, __, __, __],
+            [__, __, __, __, __, __, __, B2],
+            [__, __, __, __, __, __, __, A2],
+        ], 4);
+        testUtils.setupState(state);
+        // when clicking on A2 and then somewhere else than the first line
+        // then the move is not legal
+        await testUtils.expectClickSuccess('#click_7_0');
+        await testUtils.expectClickFailure('#click_2', DiamFailure.MUST_SHIFT_TO_NEIGHBOR());
+    }));
+    it('should forbid transferring from a piece not owned by the player', fakeAsync(async() => {
+        // given a state
+        const state: DiamState = DiamState.fromRepresentation([
+            [__, __, __, __, __, __, __, __],
+            [__, __, __, __, __, __, __, __],
+            [__, __, __, __, __, __, __, B2],
+            [__, __, __, __, __, __, __, A2],
+        ], 4);
+        testUtils.setupState(state);
+        // when clicking on B2
+        // then this is not a legal selection for a shift
+        await testUtils.expectClickFailure('#click_7_1', RulesFailure.MUST_CHOOSE_PLAYER_PIECE());
+    }));
+    it('should forbid transferring if the stack would become too high', fakeAsync(async() => {
+        // given a state
+        const state: DiamState = DiamState.fromRepresentation([
+            [__, __, __, __, __, __, __, __],
+            [A1, __, __, __, __, __, __, __],
+            [A1, __, __, __, __, __, __, B2],
+            [A1, __, __, __, __, __, __, A2],
+        ], 4);
+        testUtils.setupState(state);
+        // when clicking on B2
+        await testUtils.expectClickSuccess('#click_7_0');
+        const move: DiamMove = DiamMoveShift.fromRepresentation(new Coord(7, 3), 'clockwise');
+        // then this is not a legal selection for a shift
+        const reason: string = DiamFailure.TARGET_STACK_TOO_HIGH();
+        await testUtils.expectMoveFailure('#click_0', reason, move);
+    }));
+    it('should forbid clicking on a space without selecting a piece first', fakeAsync(async() => {
+        // given the initial state
+        // when clicking any space
+        // then it should let the user know that a piece must be selected first
+        await testUtils.expectClickFailure('#click_0', DiamFailure.MUST_SELECT_PIECE_FIRST());
+    }));
+    it('should display the number of remainig pieces', fakeAsync(async() => {
```

Comment:
display the *right* [number of] remaining pieces

Reply:
done